src/preamble.js

```diff
@@ -630,8 +630,12 @@ if (Module['locateFile']) {
   }
 #if EXPORT_ES6 && USE_ES6_IMPORT_META && !SINGLE_FILE // in single-file mode, repeating WASM_BINARY_FILE would emit the contents again
 } else {
+#if ENVIRONMENT_MAY_BE_SHELL
+  wasmBinaryFile = '{{{ WASM_BINARY_FILE }}}';
```

Comment:
Should we also check if the environment actually is shell, at runtime? otherwise this can affect `node,shell` builds that always run in node.

Reply:
Done